lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -2371,11 +2373,21 @@ void ChargePointImpl::log_status_notification(UploadLogStatusEnumType status, in
 }
 
 void ChargePointImpl::signed_firmware_update_status_notification(FirmwareStatusEnumType status, int requestId) {
-    EVLOG_debug << "Sending FirmwareUpdateStatusNotification";
+    EVLOG_debug << "Sending FirmwareUpdateStatusNotification with status"
+                << conversions::firmware_status_enum_type_to_string(status);
     SignedFirmwareStatusNotificationRequest req;
     req.status = status;
     req.requestId = requestId;
 
+    if (status == FirmwareStatusEnumType::SignatureVerified) {
+        this->firmware_update_is_pending = true;
+    } else if (status == FirmwareStatusEnumType::InstallationFailed ||
+               status == FirmwareStatusEnumType::DownloadFailed ||
+               status == FirmwareStatusEnumType::InstallVerificationFailed ||
+               status == FirmwareStatusEnumType::InvalidSignature) {
+        this->firmware_update_is_pending = false;
+    }
```

Comment:
Use `switch` with enums. With that the compiler gives warnings on missing unhandled values. Try to avoid using the `default` case as it deactivates this functionality with the usual `-Wswitch` (which is in `-Wall`, but not `-Wswitch-enum`, at least in [`gcc`](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wswitch-enum)).

Reply:
Those two cases are not exhaustive (on purpose) though (so I would need a default case that just breaks to indeed avoid the warning). I added a comment in the code as explanation (of course, I could still also switch to a switch)